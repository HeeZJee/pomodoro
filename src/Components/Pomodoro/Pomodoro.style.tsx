import styled from "styled-components";

export const Wrapper = styled.div`
  width: 50vw;
  height: 70vh;
  display: flex;
  background-color: #efefef;
  border-radius: 10px;
  box-shadow: 0 0 50px 0 rgba(0, 0, 0, 0.5);

  @media screen and (max-width: 960px) {
    width: 70vh;
  }

  @media screen and (max-width: 510px) {
    width: 60vh;
  }
  @media screen and (max-width: 400px) {
    width: 50vh;
  }
`;
