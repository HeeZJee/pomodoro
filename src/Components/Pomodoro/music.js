const songsCollections = [
  "http://www.music.helsinki.fi/tmt/opetus/uusmedia/esim/a2002011001-e02.wav",
  "http://www.noiseaddicts.com/samples_1w72b820/2563.mp3",
  "http://www.noiseaddicts.com/samples_1w72b820/2558.mp3",
];

const getRandomFromRange = (min, max) => {
  return Math.floor(Math.random() * (max - min) + min);
};

const randomSongIndex = getRandomFromRange(0, songsCollections.length);

console.log("Random song index is ", randomSongIndex);

const randomSong = songsCollections[randomSongIndex];
new Audio(randomSong).play();
