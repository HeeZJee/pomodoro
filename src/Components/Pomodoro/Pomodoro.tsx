import React, { useState } from "react";
import Timer from "../TImer/Timer";
import { Wrapper } from "./Pomodoro.style";

interface iState {
  minute: number;
  second: number;
  isOn: boolean;
}

const initialState = {
  minute: 25,
  second: 0,
  isOn: false,
};

const Pomodoro = () => {
  const [state, setState] = useState<iState>(initialState);
  const { minute, second, isOn } = state;

  return (
    <Wrapper data-testid="pomodoro">
      <Timer minute={minute} second={second} />
    </Wrapper>
  );
};

export default Pomodoro;
