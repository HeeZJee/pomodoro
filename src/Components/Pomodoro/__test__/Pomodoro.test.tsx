import React from "react";
import { render } from "@testing-library/react";
import Pomodoro from "./../Pomodoro";

it("renders Pomodoro", () => {
  const { container } = render(<Pomodoro />);
  expect(container.childElementCount).toBeGreaterThanOrEqual(1);
});

it("snapshot Pomodoro", () => {
  const { getByTestId } = render(<Pomodoro />);
  expect(getByTestId("pomodoro")).toMatchSnapshot();
});
