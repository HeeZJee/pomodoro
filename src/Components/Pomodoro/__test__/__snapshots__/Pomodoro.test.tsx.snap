// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Pomodoro 1`] = `
<div
  class="sc-bdfBwQ eVglnJ"
  data-testid="pomodoro"
>
  <div>
    <span>
      25
    </span>
    :
    <span>
      0
    </span>
  </div>
</div>
`;
